File: 06_Linked_List/01_Reverse_Linked_List.java
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null || head.next == null)
            return head;
        ListNode trav = head, prev = null, next = null;

        while (trav != null) {
            next = trav.next;
            trav.next = prev;
            prev = trav;
            trav = next;
        }
        return prev;
    }
}

File: 06_Linked_List/01_Reverse_Linked_List.cpp
class Solution
{
public:
    ListNode *reverseList(ListNode *head)
    {
        if (!head || !head->next)
            return head;

        ListNode *trav = head, *prev = NULL, *next = NULL;

        while (trav)
        {
            next = trav->next;
            trav->next = prev;
            prev = trav;
            trav = next;
        }

        return prev;
    }
};